scripts/apps/ingest/directives/IngestSourcesContent.ts

```diff
@@ -580,6 +580,16 @@ export function IngestSourcesContent(ingestSources, notify, api, $location,
                     return feedingService ? feedingService.templateUrl : '';
                 };
 
+                /**
+                 * Do URL request specified in url_request field
+                 * @param provider ingest provider metadata
+                 * @param field url_request field metadata
+                 */
+                $scope.doUrlRequest = (provider: Object, field: Object): void => {
```

Comment:
Can you use the exact types for provider/field? Typing it as `Object` isn't really more useful than not specifying a type at all.

I would also drop the comments on parameters since they don't really tell much more than "this is a variable and it contains data".

Reply:
Did follow what is done in the rest of the file for consistency, but sure I can remove.
About `field` there is no type for that, it's just description dict from the backend, that why it's not specified more than that.

Reply:
> About `field` there is no type for that.

You are reading a `url` property from that variable and performing a `replace` operation which means it's a string. You could specify then `field: {url: string}`.

It doesn't matter how is it described in the back-end or even doesn't have types defined there - we can fire up the debugger and inspect which fields does the object contain - even if they are optional and we can type them as such - `{optionalProperty?: any}`